nix/overlays/packages.nix

```diff
@@ -0,0 +1,105 @@
+self: super: {
+  haskell = super.haskell // {
+    packages = super.haskell.packages // {
+      ${super.ghcName} = super.haskell.packages.${super.ghcName}.override {
+        overrides = hself: hsuper: {
+          # NOTE: avoiding IFD's/automatically generated derivations in favor of cabal2nix generated default.nix
+          aeson-ordered = hsuper.callPackage ../../server/lib/aeson-ordered { };
+          api-tests = hsuper.callPackage ../../server/lib/api-tests { };
+          # TODO: some typed hole errors ATM + tests don't compile as extra-source-files are missing
+          dc-api = super.haskell.lib.dontCheck
+            (hsuper.callPackage ../../server/lib/dc-api { });
+          hasura-error-message =
+            hsuper.callPackage ../../server/lib/error-message { };
+          graphql-parser =
+            hsuper.callPackage ../../server/lib/graphql-parser-hs {
+              hedgehog = hself.hedgehog_1_2;
+            };
+          hasura-prelude =
+            hsuper.callPackage ../../server/lib/hasura-prelude { };
+          pg-client-hs = super.haskell.lib.dontCheck
+            (hsuper.callPackage ../../server/lib/pg-client-hs {
+              resource-pool = hself.hasura-resource-pool;
+            });
+
+          hasura-schema-parsers =
+            hsuper.callPackage ../../server/lib/schema-parsers { };
+          test-harness = hsuper.callPackage ../../server/lib/test-harness { };
+
+          graphql-server = super.haskell.lib.justStaticExecutables
+            ((super.haskell.lib.dontCheck (hsuper.callPackage ../../server {
+              hedgehog = hself.hedgehog_1_2;
+              immortal = hself.immortal_0_2_2_1;
+              pg-client = hself.pg-client-hs;
+              odbc = hself.hasura-odbc;
+              resource-pool = hself.hasura-resource-pool;
+            })));
+
+          # FIXME: for remote repos we have to calculate their default.nix until they get upstreamed/maintained
```

Comment:
These seem legit. I suggest moving the actual fetches into the inputs in _flake.nix_ though, so they're pinned with _flake.lock_. (You can still specify the exact commit ref.)

Reply:
Yup, that's the plan. I need to do this anyway to not use [IFDs](https://nixos.wiki/wiki/Import_From_Derivation).